### PR TITLE
ExecutionOptions.UserContext: object -> IDictionary<string, object>

### PR DESCRIPTION
**Pull request by sungam3r**
draft fix for #217

### CODE REVIEW

File: src/Transports.Subscriptions.Abstractions/MessageHandlingContext.cs [hunk 1]
```diff
@@ -1,10 +1,11 @@
 ﻿using System;
 using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace GraphQL.Server.Transports.Subscriptions.Abstractions
 {
-    public class MessageHandlingContext : IDisposable
+    public class MessageHandlingContext : Dictionary<string, object>, IDisposable
```
Comment (sungam3r) [inline src/Transports.Subscriptions.Abstractions/MessageHandlingContext.cs]: Not sure about that. Inside there is already a `ConcurrentDictionary` for some properties. Not yet thoroughly understood this repository.

**Comment by pekkah** [comment]
Because originally user context was object we needed to hijack it so we could add more properties into it. That's why on server side it already is a dictionary. The original user context is injected into it and available on resolvers.

**Comment by sungam3r** [comment]
Need to make some more fixes?

**Comment by joemcbride** [comment]
@sungam3r Would you mind fixing this merge conflict and I'll get this merged?  There may be some other refactoring that we need to do for subscriptions, though this should at least get the core project working.

**Comment by sungam3r** [comment]
Yes, I want to do it, but I did not understand the comment from @pekkah 
> Because originally user context was object we needed to hijack it so we could add more properties into it. That's why on server side it already is a dictionary. The original user context is injected into it and available on resolvers.

I suspect I'm doing something wrong. I will come back to this a little later, I need to figure it out.

**Comment by joemcbride** [comment]
> I suspect I'm doing something wrong.

No, it was how it was designed.  Subscriptions needed to pass along some additional metadata.  @pekkah accomplished that with the `UserContext`.  But, that would override the existing `UserContext`.  So he devised a way to work around that.  Now that `UserContext` is a Dictionary this can be refactored.  I can do that work, I just want to get what you have accomplished so far committed.

**Comment by sungam3r** [comment]
You can do it yourself if it is required as quickly as possible. I still need some time to figure it out myself. I do not want to commit things if I do not fully understand what I have done. Now honestly, I am very tired, I can only return to this tomorrow.

**Comment by joemcbride** [comment]
No rush.  In fact, it looks like there is only one file conflicting.  I'll take care of it.

**Comment by joemcbride** [comment]
@sungam3r good news, subscriptions still work with those changes.  🎉

**Comment by sungam3r** [comment]
Good

**Pull request merged**